runtime/lua/vim/lsp/structures/versioned_text_document_identifier.lua

```diff
@@ -0,0 +1,13 @@
+local VersionedTextDocumentIdentifier = {}
+
+local buf_versions = {}
+
+VersionedTextDocumentIdentifier.buf_get_version = function(bufnr)
+  return buf_versions[bufnr] or 0
+end
+
+VersionedTextDocumentIdentifier.buf_set_version = function(bufnr, version)
+  buf_versions[bufnr] = version
+end
```

Comment:
I also thought that'd make sense and I was worried about a client with a lower version getting its edit messages discarded. But since we change the version on every buffer change (if I'm reading the following line of code correctly), a single version number seems to make sense, right? https://github.com/neovim/neovim/blob/570c60c6507e88addb9f131d3be4dd56949aff75/runtime/lua/vim/lsp.lua#L599 If client A changes version 1 of the buffer, we need to increment the version and send the `textDocument/didChange` to both servers through client A and B, right?

Reply:
Yes, we want to share the version between them so that they can't apply changes that don't make sense anymore.